packages/transformer/src/IModelTransformer.ts

```diff
@@ -1930,12 +1944,24 @@ export class IModelTransformer extends IModelExportHandler {
    * This override calls [[onTransformModel]] and then [IModelImporter.importModel]($transformer) to update the target iModel.
    */
   public override onExportModel(sourceModel: Model): void {
-    if (IModel.repositoryModelId === sourceModel.id) {
+    if (
+      this._options.skipPropagateChangesToRootElements &&
+      IModel.repositoryModelId === sourceModel.id
+    )
       return; // The RepositoryModel should not be directly imported
-    }
     const targetModeledElementId: Id64String = this.context.findTargetElementId(
       sourceModel.id
     );
+    if (
+      sourceModel.id === IModel.repositoryModelId &&
+      targetModeledElementId != sourceModel.id
+    ) {
+      /**
+       * The repositoryModel should not be directly imported if its targetModeledElementId is different than the sourceModel.id, since this means the repositoryModel/rootSubject was remapped.
+       * There can only be one repositoryModel per database so we will return early and not import it.
+       */
+      return;
+    }
```

Comment:
```suggestion
    // there can only be one repositoryModel per database, so ignore the repo model on remapped subjects
    const isRemappedRootSubject = sourceModel.id === IModel.repositoryModelId && targetModeledElementId != sourceModel.id;
    if (isRemappedRootSubject) return;
```

I was not aware that there can only be one repository model... interesting decision with potential ramifications for forks I suppose?